### PR TITLE
rt-scope should be evaluated after rt-if, and inside each rt-repeat iteration

### PR DESCRIPTION
**Pull request by anchann**
Hello! I have a number of PRs, and wasn't sure where to put the high level description, so I'll copy and paste it in all of them.

Coming from Angular, I found the react world to be a sad place as far as DSLs go — JSX is a sorry excuse that for some reason beyond me is accepted by the community as the standard. And then I stumbled upon your react-templates work. Thank you!! While converting my work over, I came across a number of wishlist features, and today finally decided to take some time to implement them. They are all aimed at improving readability/writability, as well as sometimes avoiding unnecessary nested elements.

===

The two top commits in this stack of three rely on one submitted as a separate PR. Couldn't find a way to submit them as a PR on top of another submitted PR... we can deal with rebases if you think the changes look good.

Basically all these deal with the semantics of rt-scope. The details are in each commit message.

### CODE REVIEW

**Comment by gregwebs** [comment]
@avi @idok This is a great project and I would like to see it flourish outside of Wix. The website and demo are fantastic.

But as an open source project it needs continued stewardship. The minimum effort is just to set a deadline for when a pull request will be reviewed and give updates as any deadlines are missed.
So let us know when we can expect PRs to be reviewed.

**Comment by bjornlll** [comment]
Has this PR been reviewed yet? Any timeline of when we'll see it make its way into react-templates?

**Comment by gregwebs** [comment]
First commit has already been taken. I could really use the last one.

**Comment by idok** [comment]
Sorry for the delay.
This commit is breaking backward compatibility. We cannot accept it as it is as we have a very large code base that will break.
All other PR were merged.

**Comment by anchann** [comment]
Perhaps we can separate the issue of backwards compatibility for your codebase and the issue of what the best semantics are for all future users of this DSL? If we discuss the latter, and agree that the changes make sense, a major version bump together with some upgrade notes would be enough to communicate to existing users that there are breaking changes.

**Comment by idok** [comment]
Then lets make it a breaking change, please create a new PR or change this one and change the semantics of rt-scope.
Thanks

**Comment by anchann** [comment]
Here you go: https://github.com/wix/react-templates/pull/62

Decided to submit this as a stand-alone feature. The maintainers can make the call on whether to drop existing rt-scope and simply rename rt-inner-scope to rt-scope, thus adopting the new semantics in the breaking manner.
